tests/basic/podmetric_test.go

```diff
@@ -70,9 +79,16 @@ var _ = Describe("{PodMetricFunctional}", func() {
 				time.Sleep(waitDuration)
 
 				log.InfoD("Check metering data is accurate")
-				meteringData, err = getMeteringData(clusterUUID, meteringInterval)
-				log.FailOnError(err, "Failed to get metering data")
-				existsData = len(meteringData) > 0
+
+				// try to get non-empty metering data for 3 minutes
+				Eventually(func() bool {
+					meteringData, err = getMeteringData(clusterUUID, meteringInterval)
```

Comment:
What happens if this is not ture eventually, will it still execute next lines.
If not can you use our internal method `task.DoRetryWithTimeout`

Reply:
Eventually is a ginkgo built-in function that will retry until a function is true.

The test will fail if that is not the case, which is the goal here